module/VuFind/src/VuFind/Hierarchy/TreeRenderer/JSTree.php

```diff
@@ -220,10 +220,10 @@ protected function getContextualUrl($node, $context)
             if ('collection' === $type && !$this->collectionsEnabled) {
                 $type = 'record';
             }
-            $url = $this->getUrlFromRouteCache($type, $node->id);
+            $url = $this->getUrlFromRouteCache($type, urlencode($node->id));
             return $type === 'collection'
                 ? $url . '#tabnav'
-                : $url . '#tree-' . preg_replace('/\W/', '-', $node->id);
+                : $url . '#tree-' . preg_replace('/\W/', '-', urlencode($node->id));
```

Comment:
Have you confirmed that this works as expected? Is url-encoding necessary in hash fragments? When I put a URL-encoded character in my hash and then `console.log window.location.hash;` in my browser, it does not appear that the encoded character is getting decoded... which makes me wonder whether the correct anchor will get matched up if we do this encoding.

Reply:
Yes, it seems that the hash-part does not need encoding. Or at least the hierarchytree is scrolled to the right position so that the opened record is visible on the screen. Tested with this kind of id: